redbot/core/core_commands.py

```diff
@@ -235,8 +236,18 @@ def _get_package_strings(self, packages: list, fmt: str, other: tuple = None):
         str
             Invite URL.
         """
-        app_info = await self.bot.application_info()
-        return discord.utils.oauth_url(app_info.id)
+        if self.bot.user.bot:
+            app_info = await self.bot.application_info()
+            perms_int = await self.bot.db.invite_perm()
+            redirect = await self.bot.db.invite_redirect()
+            permissions = discord.Permissions(perms_int)
+            return discord.utils.oauth_url(app_info.id, permissions, redirect_uri=redirect)
+        return "Not a bot account!"
```

Comment:
This special casing for userbots should be removed. We dont support user bots, they are against discord's TOS.